src/EventHub/EventHub/Cmdlets/NetworkRuleSet/SetAzureEventHubNetworkrule.cs

```diff
@@ -0,0 +1,115 @@
+    public class SetAzureEventHubNetworkrule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }       
+
+        [Parameter(Mandatory = false, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "Default Action for NetwrokeuleSet")]
+        [ValidateSet("Allow", "Deny", IgnoreCase = true)]
+        [PSArgumentCompleter("Allow", "Deny")]
+        [PSDefaultValue(Value ="Deny")]
+        public string DefaultAction { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 3, HelpMessage = "List of IPRuleSet")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNWRuleSetIpRulesAttributes> IPRules { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "List of VirtualNetworkRules")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNWRuleSetVirtualNetworkRulesAttributes> VirtualNtewrokRules { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "NetworkruleSet Configuration Object")]
+        [ValidateNotNullOrEmpty]
+        public PSNetworkRuleSetAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "Resource ID of Namespace")]
```

Comment:
@v-Ajnava this should be `ValueFromPipelineByPropertyName = true` instead of `ValueFromPipeline = true`

Reply:
Resolved